### PR TITLE
chore: update numaflow-transformer-config.yaml

### PR DESCRIPTION
**Pull request by jy4096**
fix: https://github.com/numaproj/numaflow/issues/1530

Test `images`
pipeline.yaml
```yaml
apiVersion: numaflow.numaproj.io/v1alpha1
kind: Pipeline
metadata:
  name: simple-pipeline
spec:
  sideInputs:
    - name: dummy
      container:
        image: numaflow-go:latest
        args:
          - "side-inputs-example"
      trigger:
        schedule: "*/15 * * * *"
        # timezone: America/Los_Angeles
  vertices:
    - name: in
      source:
        udsource:
          container:
            image: numaflow-go:latest
            args:
              - "udsource-example"
      sideInputs:
        - dummy
    - name: forward
      scale:
        min: 1
      sidecars:
        - name: my-sidecar
          image: busybox:latest
          command: [ "/bin/sh", "-c", "echo \"my-sidecar is running!\" && tail -f /dev/null" ]
      udf:
        container:
          image: numaflow-python:latest
          args:
            - python
            - starter.py
            - "forward-map"
    - name: out
      sink:
        udsink:
          container:
            image: numaflow-go:latest
            args:
              - "sink-example"
  edges:
    - from: in
      to: forward
    - from: forward
      to: out
```
kustomization.yaml
```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

configurations:
- numaflow-transformer-config.yaml

resources:
- pipeline.yaml

images:
- name: numaflow-go
  newTag: foo
- name: numaflow-python
  newTag: bar
- name: busybox
  newTag: test
```

run `kustomize build --load_restrictor=none . > ~/Desktop/manifest.yaml`

got
```yaml
apiVersion: numaflow.numaproj.io/v1alpha1
kind: Pipeline
metadata:
  name: simple-pipeline
spec:
  edges:
  - from: in
    to: forward
  - from: forward
    to: out
  sideInputs:
  - container:
      args:
      - side-inputs-example
      image: numaflow-go:foo
    name: dummy
    trigger:
      schedule: '*/15 * * * *'
  vertices:
  - name: in
    sideInputs:
    - dummy
    source:
      udsource:
        container:
          args:
          - udsource-example
          image: numaflow-go:foo
  - name: forward
    scale:
      min: 1
    sidecars:
    - command:
      - /bin/sh
      - -c
      - echo "my-sidecar is running!" && tail -f /dev/null
      image: busybox:test
      name: my-sidecar
    udf:
      container:
        args:
        - python
        - starter.py
        - forward-map
        image: numaflow-python:bar
  - name: out
    sink:
      udsink:
        container:
          args:
          - sink-example
          image: numaflow-go:foo
```

### CODE REVIEW

File: docs/user-guide/reference/kustomize/numaflow-transformer-config.yaml [hunk 1]
```diff
@@ -52,6 +70,10 @@ nameReference:
     kind: Pipeline
   - path: spec/templates/daemon/initContainerTemplate/envFrom/configMapRef/name
     kind: Pipeline
+  - path: spec/templates/daemon/sideInputsContainerTemplate/env/valueFrom/configMapKeyRef/name
```
Comment (whynowy) [inline docs/user-guide/reference/kustomize/numaflow-transformer-config.yaml:73]: Are these 2 lines right?

File: docs/user-guide/reference/kustomize/numaflow-transformer-config.yaml [hunk 2]
```diff
@@ -111,6 +145,10 @@ nameReference:
     kind: Pipeline
   - path: spec/templates/daemon/initContainerTemplate/envFrom/secretRef/name
     kind: Pipeline
+  - path: spec/templates/daemon/sideInputsContainerTemplate/env/valueFrom/secretKeyRef/name
```
Comment (whynowy) [inline docs/user-guide/reference/kustomize/numaflow-transformer-config.yaml:148]: ?

File: docs/user-guide/reference/kustomize/numaflow-transformer-config.yaml [hunk 3]
```diff
@@ -212,26 +304,52 @@ varReference:
...
+- path: spec/sideInputs/container/env/value
+  kind: Pipeline
 - path: spec/vertices/containerTemplate/env/value
   kind: Pipeline
 - path: spec/vertices/initContainerTemplate/env/value
   kind: Pipeline
-- path: spec/vertices/initContainers/env/value
+- path: spec/vertices/sideInputsContainerTemplate/env/value
   kind: Pipeline
-- path: spec/vertices/sidecars/env/value
```
Comment (jy4096) [inline docs/user-guide/reference/kustomize/numaflow-transformer-config.yaml:221]: moved to line 287

File: docs/user-guide/reference/kustomize/numaflow-transformer-config.yaml [hunk 4]
```diff
@@ -212,26 +304,52 @@ varReference:
...
+  kind: Pipeline
+- path: spec/sideInputs/container/command
+  kind: Pipeline
+- path: spec/sideInputs/container/env/value
+  kind: Pipeline
 - path: spec/vertices/containerTemplate/env/value
   kind: Pipeline
 - path: spec/vertices/initContainerTemplate/env/value
   kind: Pipeline
-- path: spec/vertices/initContainers/env/value
```
Comment (jy4096) [inline docs/user-guide/reference/kustomize/numaflow-transformer-config.yaml:219]: moved to line 325

**Review by whynowy — APPROVED** [review]
LGTM, tested?

**Comment by jy4096** [comment]
> LGTM, tested?

tested the `images` and passed.

**Pull request merged**